flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/functions/casting/ArrayToArrayCastRule.java

```diff
@@ -109,7 +110,7 @@ protected String generateCodeBlockInternal(
 
     private static String arrayElementType(LogicalType t) {
         if (t.isNullable()) {
```

Comment:
use if (t.isNullable() && !CodeGenUtils.isPrimitiveNullable(t))?
when t is primitiveNullable, it would be better to use primitive type

Reply:
if it's nullable, you can't use the primitive array as you can't add nulls. Maybe I didn't get what you mean

Reply:
IMO, CodeGenUtils.isPrimitiveNullable decides whether a  primitiveType is nullable, if there is a type supports CodeGenUtils.isPrimitiveNullable = true in the future, we should use primitive type here. 

Maybe it is impossible to having such kind of type, you can ignore this comment.

Reply:
Ok now I see what you mean, but this doesn't make much difference in the end from the code here, right?

Reply:
yes, it is no difference in the generated code now, let's go ahead.